datasets/wikisource/wikisource.py

```diff
@@ -0,0 +1,236 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Wikisource dataset."""
+
+
+import bz2
+import codecs
+import json
+import re
+import xml.etree.ElementTree as etree
+
+import datasets
+
+
+logger = datasets.logging.get_logger(__name__)
+
+
+_CITATION = """\
+@ONLINE {wikidump,
+    author = {Wikimedia Foundation},
+    title  = {Wikimedia Downloads},
+    url    = {https://dumps.wikimedia.org}
+}
+"""
+
+_DESCRIPTION = """\
+Wikisource dataset containing cleaned articles for all languages.
+"""
+
+_HOMEPAGE = "https://dumps.wikimedia.org"
+
+_LICENSE = "Creative Commons Attribution-ShareAlike 3.0 Unported"
+
+_BASE_URL_TMPL = "https://dumps.wikimedia.org/{lang}wikisource/{date}/"
+_INFO_FILE = "dumpstatus.json"
+
+WIKISOURCE_LANGUAGES = []
```

Comment:
FYI if you're looking to build a static list of language codes, these statistics might be of help for guiding that choice: https://meta.wikimedia.org/wiki/Wikisource#List_of_Wikisources (where the `good` column is the one to pay attention to as # of content pages)

Reply:
DONE @geohci.